src/dp_abstraction/aca_dataplane_ovs.cpp

```diff
@@ -438,15 +447,25 @@ int ACA_Dataplane_OVS::update_neighbor_state_workitem(NeighborState current_Neig
               // we can consider doing this L2 neighbor creation as an on demand rule to support scale
               // when we are ready to put the DVR rule as on demand, we should put the L2 neighbor rule
               // as on demand also
+  
+              if(current_NeighborState.operation_type() == OperationType::CREATE){
+                rc = ACA_ARP_Responder::get_instance().add_arp_entry(&stArpCfg);
```

Comment:
I think it will be better to do the add/update arp entry call inside:
ACA_Vlan_Manager::get_instance().create_l2_neighbor

ACA_OVS_L2_Programmer::get_instance().create_or_update_l2_neighbor will call into that.